processing/src/main/java/org/apache/druid/error/DruidException.java

```diff
@@ -176,6 +176,21 @@ public static DruidException defensive(String format, Object... args)
     return defensive().build(format, args);
   }
 
+  /**
+   * Build a "defensive" exception, this is an exception that should never actually be triggered. Throw to
+   * allow messages to be seen by developers
+   *
+   * @param condition - boolean condition to validate
+   * @param msg - passed through to InvalidInput.exception()
+   * @param args - passed through to InvalidInput.exception()
+   */
+  public static void conditionalDefensive(boolean condition, String msg, Object... args)
```

Comment:
Probably need to add @SuppressWarnings("unused") or find a place to use it.  It's a bit of a chicken-and-egg as I'm sure these will eventually be used...